phpBB/phpbb/avatar/driver/remote.php

```diff
@@ -84,6 +84,24 @@ public function process_form($request, $template, $user, $row, &$error)
 			return false;
 		}
 
+		/**
+		 * Event to make custom validation of avatar upload
+		 *
+		 * @event core.ucp_profile_avatar_upload_validation
+		 * @var	string	$url	Image url
+		 * @var	string	$width	Image width
+		 * @var	string	$height	Image height
+		 * @var	array	$error	Error message array
+		 * @since 3.2.6-RC1
+		 */
+		$vars = array('url', 'width', 'height', 'error');
+		extract($phpbb_dispatcher->trigger_event('core.ucp_profile_avatar_upload_validation', compact($vars)));
+
+		if (!empty($error))
```

Comment:
It looks redundant this block, I think you should now remove the lines

https://github.com/phpbb/phpbb/blob/fe09a8a4b3c75babdbd21729491a35bd7f861a94/phpBB/phpbb/avatar/driver/remote.php#L82-L85

Reply:
this block is for show error from custom validation

Reply:
The errors from custom validation should be added to the existing array.

Reply:
they are added inside the hook.

example:

`
class listener implements EventSubscriberInterface 
{

	static public function getSubscribedEvents()
	{
		return array(
			'core.ucp_profile_avatar_upload_validation'     => 'avatar_upload_validation',
		);
	}

    public function avatar_upload_validation($event)
    {
        if ($event['url'] === 'http://test.com/image.jpg')
        {
            $event['error'] = array_merge($event['error'], array('test.com is not a valid image hosting!'));
        }
    }
`

Like the rest of the validators of the file when it finds an error the function returns false

Reply:
```
			$error = $event['error'];
			if ($event['url'] === 'http://test.com/image.jpg')
			{
				$error[] = $this->language->lang('NOT_A_VALID_IMG_HOSTING');
			}
			$event['error'] = $error;
```
or use the update_subarray() method.

Reply:
Also, `$phpbb_dispatcher` is not available https://travis-ci.org/phpbb/phpbb/jobs/520054536#L1357

Reply:
fixed